action_server_bt/src/action_server_bt.cpp

```diff
@@ -116,22 +122,28 @@ void ActionServerBT::execute(const std::shared_ptr<GoalHandleActionTree> goal_ha
       {
         action_result->error_message = "Action Server canceling and halting Behavior Tree";
         RCLCPP_ERROR(kLogger, action_result->error_message.c_str());
-        tree.haltTree();
+        tree_->haltTree();
         goal_handle->canceled(action_result);
+        onTreeExecutionCompleted(status, true);
         return;
       }
 
-      user_cbs_.on_loop(*global_blackboard);
       // tick the tree once and publish the action feedback
-      status = tree.tickExactlyOnce();
+      status = tree_->tickExactlyOnce();
+
+      if (!onLoopAfterTick(status))
+      {
+        break;
```

Comment:
Don't we need to `halt` the tree here? I think it also makes sense to return from the action goal that it was aborted...
```
action_result->node_status = convert_node_status(status);
action_result->error_message = "Action Server aborted in onLoopAfterTick()";
RCLCPP_ERROR(kLogger, action_result->error_message.c_str());
goal_handle->abort(action_result);
onTreeExecutionCompleted(status, true);
```

Reply:
check my latest commit